lib/adapters/api.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+let juttle_utils = require('../runtime/juttle-utils');
+
+//
+// The AdapterAPI is the exported interface to the juttle compiler / runtime
+// that is expected to be used by all adapters.
+//
+// The version field corresponds to the juttle version in which the latest
+// changes were made to the adapter API.
+//
+let AdapterAPI = {
+    version: '0.5.0',
+    AdapterRead: require('./adapter-read'),
+    AdapterWrite: require('./adapter-write'),
+    compiler: {
+        ASTVisitor: require('../compiler/ast-visitor'),
+        FilterJSCompiler: require('../compiler/filters/filter-js-compiler')
+    },
+    errors: require('../errors'),
```

Comment:
For the most part adapters can just use the methods attached to the read and write base classes, but there may be cases in which the error needs to be thrown from a global context such as when validating config loading the adapter or from a utility module. 

-m


> On Feb 16, 2016, at 6:38 PM, Mark Stemm <notifications@github.com> wrote:
> 
> In lib/adapters/api.js:
> 
> > +//
> > +// The AdapterAPI is the exported interface to the juttle compiler / runtime
> > +// that is expected to be used by all adapters.
> > +//
> > +// The version field corresponds to the juttle version in which the latest
> > +// changes were made to the adapter API.
> > +//
> > +let AdapterAPI = {
> > +    version: '0.5.0',
> > +    AdapterRead: require('./adapter-read'),
> > +    AdapterWrite: require('./adapter-write'),
> > +    compiler: {
> > +        ASTVisitor: require('../compiler/ast-visitor'),
> > +        FilterJSCompiler: require('../compiler/filters/filter-js-compiler')
> > +    },
> > +    errors: require('../errors'),
> Do we need errors at all, or can all adapters just throw using this.xxxError?
> 
> —
> Reply to this email directly or view it on GitHub.
>